tests/unit/test_autoclaved.py

```diff
@@ -44,23 +55,46 @@
         'intra_off': 1536,
         'report_size': 3118,
         'expected_shasum': 'e13999959d636ad2b5fd8d50493a257a2c616b0adad086bf7211de5f09463f6d'
-    }
+    },
+
+    # Four-frame report starting and ending at the middle of the frame
+    'multi_frame_mid_offset': {
+        'autoclaved_filename': '2013-05-06/tcp_connect.0.tar.lz4',
+        'textname': '2013-05-06/20130506T113818Z-VN-AS45899-tcp_connect-no_report_id-0.1.0-probe.yaml',
+        'frame_off': 210860,
+        'total_frame_size': 264195 + 17743 - 210860,
+        'intra_off': 2381,
+        'report_size': 865023,
+        'expected_shasum': 'dc5e65ae1785afac7bb2d5c01b7d052954236c6e5a4e0bee38d0693cc52285bf',
+    },
+
+    # It's hard to find test-cases ending at the edge of non-last frame because
+    # of tar padding and headers, but here is something like that.
+    'next_frame_newline': {
+        'autoclaved_filename': '2016-04-18/http_requests.06.tar.lz4',
+        'textname': '2016-04-18/20160417T080443Z-GB-AS5607-http_requests-no_report_id-0.1.0-probe.yaml',
+        'frame_off': 555968,
+        'total_frame_size': 86597, # single frame, all alike reports fit into single frame :-/
+        'intra_off': 211073,
+        'report_size': 151041,
+        'expected_shasum': '58dbc014da07fb3192e2231be37e1acc7344cf87ecc8afabf84bf5c387d1a825',
+    },
 }
 
-def test_decompress(client):
-    for name, ac in AUTOCLAVED.items():
-        decompressor = decompress_autoclaved(
-                ac['autoclaved_filename'],
-                ac['frame_off'],
-                ac['total_frame_size'],
-                ac['intra_off'],
-                ac['report_size'])
-        download_size = 0
-        h = hashlib.sha256()
-        g = decompressor()
-        all_data = b''
-        for chunk in g:
-            all_data += chunk
-            h.update(chunk)
-        assert len(all_data) == ac['report_size'], len(all_data)
-        assert h.hexdigest() == ac['expected_shasum'], h.hexdigest()
+@pytest.mark.parametrize("name,ac", AUTOCLAVED.items())
```

Comment:
Ah neat. I didn't know `pytest` supported this (this is actually the first time I use pytest)!